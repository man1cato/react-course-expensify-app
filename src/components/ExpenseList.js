import React from 'react';
import {connect} from 'react-redux';    //STORE CONNECTOR
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => <ExpenseListItem key={expense.id} {...expense} /> )
            )
        }
    </div>
);


//THE FOLLOWING PULLS IN THE STATE FROM THE STORE AND CONNECTS IT TO THIS COMPONENT
const mapStateToProps = (state) => ({ expenses: selectExpenses(state.expenses, state.filters) });

export default connect(mapStateToProps)(ExpenseList);
