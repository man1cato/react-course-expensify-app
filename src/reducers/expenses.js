//EXPENSES REDUCER
const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_EXPENSE':
            return [...state, action.expense];      ///... operator injects all the values from the named array
        case 'REMOVE_EXPENSE':
            return state.filter((expense) => expense.id !== action.id); 
        case 'EDIT_EXPENSE':
            return state.map((expense) => expense.id === action.id ? {...expense, ...action.updates} : expense);
        default:
            return state;
    }
};

export default expensesReducer;