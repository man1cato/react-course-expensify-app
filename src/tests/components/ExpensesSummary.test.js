import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import {filters} from '../fixtures/filters';


test('should render expenses summary with count and total for a single expense', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render expenses summary with count and total for multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={7} expensesTotal={51065} />);
    expect(wrapper).toMatchSnapshot();
});