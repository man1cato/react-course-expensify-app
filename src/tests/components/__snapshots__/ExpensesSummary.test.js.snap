// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render expenses summary with count and total for a single expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $2.35
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render expenses summary with count and total for multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        7
      </span>
       
      expenses
       totalling 
      <span>
        $510.65
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
